### ISSUE TITLE
Move from __main__.py to a proper module

### ISSUE THREAD

**Issue by jkodroff** [pulumi/pulumi#11590]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

`pulumi new python` creates a file `__main__.py`. This creates problems in VS Code with the Python extension installed: Pylance complains that `__main__.py` overrides the stdlib. The entire file's contents get a yellow underline, and this creates a bad user experience for users in VS Code (which I would presume is the most popular editor for Python projects, or at least near the top).

![image](https://user-images.githubusercontent.com/410614/206469232-66f31ea1-f09f-4d83-9833-2abcdc00e3aa.png)

Per the Python experts at Pulumi (@kpitzen), we should move to a proper module.

**Comment by justinvp** [comment]
We should definitely add proper module support and move to using that in our templates. I believe we already have an issue tracking that, but would need to search for it.

In the meantime, as a quick fix, I wonder if there is some kind of config file for vs code we could include in our templates to tell pylance to ignore this.

**Comment by kpitzen** [comment]
Whatever solution we come to on this, we should ensure we preserve the current behavior by default.  It's likely fine to update `pulumi new`, but if we're using a different form of module resolution + config to run Pulumi programs, we'll need to have `__main__.py` as a fallback to avoid breaking current user programs.

**Comment by jkodroff** [comment]
AFAICT (and I am not super confident - spent like 10 minutes on it), you have to disable Pylance altogether, at least for the whole `__main__.py` file. I haven't found an example of disabling the specific warning we're seeing.

**Comment by kpitzen** [comment]
For now, we can set this:
<img width="459" alt="image" src="https://user-images.githubusercontent.com/4128006/206473368-9e3055ab-aa0f-448c-a34f-b3a5c89d627d.png">

**Comment by jkodroff** [comment]
@kpitzen I am very glad to stand corrected! Thank you!

**Comment by devopsangel** [comment]
I ended up adding `project.toml`. 

```
[tool.pyright]

venv = ".venv"

reportMissingImports = true
reportShadowedImports = false

pythonVersion = "3.10"
```

**Comment by codethief** [comment]
I find the naming of the entrypoint as `__main__.py` rather surprising. As per https://docs.python.org/3/library/__main__.html, `__main__.py` has a very specific purpose: It is used for executing a _package_ (`python -m foo`). To this end, `__main__.py` is placed _inside_ that package `foo`, like so:

```
|- foo/
|  |- __main__.py
…
``` 

Then, upon executing `python -m foo` (e.g. from within the parent directory of `foo`), the Python interpreter will correctly resolve absolute imports of `foo.*` inside `foo/__main__.py` and any other modules that are being loaded. 

However, it seems Pulumi will invoke `__main__.py` via something like `python foo/__main__.py` which will *not* add the parent folder of `foo` to `sys.path`, but instead will add `foo` itself to `sys.path`. In other words, absolute imports will end up being resolved relative to `foo/`, not relative to the parent folder of `foo`, and relative imports in `__main__.py` will not work ([see also](https://stackoverflow.com/questions/4042905/what-is-main-py/64787998#64787998)).

Long story short: Man, this is confusing. Please name it `main.py` like everyone else or let people choose what & where the entrypoint should be.
